x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -98,23 +144,80 @@ export const DetailsPageMappingsContent: FunctionComponent<{
 
   useMappingsStateListener({ value: parsedDefaultValue, status: 'disabled' });
 
-  const {
-    fields: { byId, rootLevelFields },
-    search,
-    documentFields,
-  } = useMappingsState();
+  const onCancelAddingNewFields = useCallback(() => {
+    setAddingFields(!isAddingFields);
+
+    //  reset mappings to previous state
+    dispatch({
+      type: 'editor.replaceMappings',
+      value: {
+        ...previousState,
+        documentFields: {
+          status: 'disabled',
+          editor: 'default',
+        },
+      },
+    });
+  }, [isAddingFields, dispatch, previousState]);
+
+  const addFieldButtonOnClick = useCallback(() => {
+    setAddingFields(!isAddingFields);
+
+    // when adding new field, save previous state. This state is then used by FieldsList component to show only saved mappings.
+    setPreviousStateFields(getFieldsFromState(state));
+    setPreviousState(state);
```

Comment:
Save state to `previousState` when adding a new field, which would then be sent as a props to [FieldList component](https://github.com/elastic/kibana/blob/445e56db08dfa0b6078e6891a05cfff8e24234f3/x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx#L245). This would enable createField to show new unsaved fields from [state](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/mappings_state_context.tsx#L68) whereas FieldList would fetch fields from `previousState`